target/linux/mediatek/patches-5.10/900-dts-mt7622-bpi-r64-aliases-for-dtoverlay.patch

```diff
@@ -78,3 +78,38 @@
 +		};
 +	};
 +};
+--- /dev/null
++++ b/arch/arm64/boot/dts/mediatek/mt7622-bananapi-bpi-r64-spi0.dts
+@@ -0,0 +1,22 @@
++/* SPDX-License-Identifier: (GPL-2.0-only OR MIT) */
++
++#include <dt-bindings/gpio/gpio.h>
++
++/dts-v1/;
++/plugin/;
++
++/ {
++	compatible = "bananapi,bpi-r64", "mediatek,mt7622";
++
++	fragment@0 {
++		target = <&spi0>;
++		__overlay__ {
++			spidev: spidev@0 {
++				compatible = "bpi,mt65xx";
++				spi-max-frequency = <1000000>;
++				reg = <0>;
++			};
++		};
++	};
++};
++
+--- a/drivers/spi/spidev.c
++++ b/drivers/spi/spidev.c
+@@ -675,6 +675,7 @@ static struct class *spidev_class;
+ 
+ #ifdef CONFIG_OF
+ static const struct of_device_id spidev_dt_ids[] = {
++	{ .compatible = "bpi,mt65xx" },
```

Comment:
Put the spi **client** device here. `bpi,mt65xx` is the spi **host**.

Reply:
Okay, maybe I'm trying to do something really stupid. I have a DW3000. I have to communicate via spi with it. I thought I just use spidev for doing communication from userspace. There was a discussion that you need to add some compatible string to [spidev driver](http://forum.banana-pi.org/t/bpi-r2-spi-communication/4779/138). Now after reading the whole discussion I'm unsure if I'm still doing it right since the actual dev for bpi said that adding some compatibility string is wrong. Now you also say that I should add my spi client device, so the dw3000?

Actually, adding `compatible = "spidev";` in the node should be enough.

Reply:
Okay, [stackoverflow](https://stackoverflow.com/a/53639989/8474618) also says I should just add my device as compatible string. Then I do it. ;)

Reply:
Or I just use https://github.com/torvalds/linux/blob/master/drivers/spi/spidev.c#L678 ?

Reply:
I guess if that DW3000 hardware requires a driver using spi-dev in user-space, you should add "qorvo,dw3000" as compatible string.